[Dai Jinghong, Nicholas] iP

nicholasdaijh opened the pull request:
# Nicholas
> ``` “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes)) ```

Nicholas frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ *SUPER* FAST to use
+ All you need to do is,

download it from [here](https://github.com/nicholasdaijh/ip/releases/tag/A-Jar).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you :grin:
4. And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

`
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nicholasdaijh committed:
changed chatbot name from "Duke" to "Nicholas"

nicholasdaijh committed:
added scanner to echo user's input

nicholasdaijh committed:
Add, list

nicholasdaijh committed:
added marked as done feature

nicholasdaijh committed:
added Mark As Done feature

nicholasdaijh committed:
added three Task types: Todos, Events, Deadlines

nicholasdaijh committed:
added error handling

nicholasdaijh committed:
added deleting function

nicholasdaijh committed:
added automated text UI testing

nicholasdaijh committed:
Add OOP and save function

nicholasdaijh committed:
Add dates and times and packages

nicholasdaijh committed:
Add oop and packages

nicholasdaijh committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

nicholasdaijh committed:
Add Gradle

nicholasdaijh committed:
Add JUnit tests

nicholasdaijh committed:
Package the App as a JAR file

nicholasdaijh committed:
Add Level-9

nicholasdaijh committed:
Add JavaDoc

nicholasdaijh committed:
Add JavaDoc again

nicholasdaijh committed:
Add Javadoc

nicholasdaijh committed:
Adjust to CodingStandard

nicholasdaijh committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Adjust to CodingStandard

nicholasdaijh committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add JavaDoc
  Add Level-9

# Conflicts:
#	src/main/java/nicholas/ui/Nicholas.java

nicholasdaijh committed:
Change content in tasks.txt

nicholasdaijh committed:
Add exceptions

nicholasdaijh committed:
Add checkstyle

minhkiwi14 commented on src/main/java/nicholas/ui/Storage.java:
```diff
@@ -0,0 +1,79 @@
+import java.util.List;
+import java.util.Scanner;
+
+import nicholas.tasks.Task;
```
I like the order of your importing!

minhkiwi14 commented on src/main/java/nicholas/ui/Parser.java:
```diff
@@ -0,0 +1,142 @@
+            }
+            return deadlineTask;
+        case "[E":
+            int eventStartIndexBegin = descriptionAndDate.indexOf("from:") + 6;
+            int eventStartIndexEnd = descriptionAndDate.indexOf("to:") - 1;
+            int eventEndIndexBegin = descriptionAndDate.indexOf("to:") + 4;
+            int eventEndIndexEnd = descriptionAndDate.length() - 1;
+            if (eventStartIndexBegin == -1 || eventStartIndexEnd == -1 || eventEndIndexBegin == -1) {
+                throw new IllegalArgumentException("Invalid event format");
+            }
+            String eventStart = descriptionAndDate.substring(eventStartIndexBegin, eventStartIndexEnd).trim();
+            String eventEnd = descriptionAndDate.substring(eventEndIndexBegin, eventEndIndexEnd).trim();
+            Task eventTask = parseEvent(descriptionAndDate.substring(0, eventStartIndexBegin - 8)
+                    + " /from " + reverseParseDate(eventStart) + " /to " + reverseParseDate(eventEnd));
+            if (input.charAt(4) == 'X') {
+                eventTask.markAsDone();
+            }
+            return eventTask;
```
Should this block be refactored out better?

minhkiwi14 commented on src/main/java/nicholas/ui/Nicholas.java:
```diff
@@ -0,0 +1,137 @@
+        // Save tasks to file before exiting
+        storage.saveTasks(taskList.getTasks());
+
+        // Show goodbye message
```
Nice comment to make the code understandable!

minhkiwi14 commented on src/main/java/nicholas/exception/EmptyCommandException.java:
```diff
@@ -0,0 +1,15 @@
+     * Constructs an EmptyCommandException with a specific command type.
+     *
+     * @param commandType The type of command that is missing a description.
+     */
```
I like these header comments!

minhkiwi14 commented on src/main/java/nicholas/tasks/Todo.java:
```diff
@@ -0,0 +1,29 @@
+     * @param description The description of the Todo task.
+     */
+    public Todo(String description) {
+        // Call parent constructor
```
Perhaps omit this comment here?

minhkiwi14 reviewed (commented):
LGTM. Just a few nits to fix.

nictjh commented on src/main/java/nicholas/tasks/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class Deadline extends Task {
+    // Deadline date or time
+    private String by;
```
whats upp broo, the variable "by" is quite vague and unclear can try "deadline" etc... so it will be more descriptive

nictjh commented on src/main/java/nicholas/tasks/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     * @return The size of the task list.
+     */
+    public int size() {
```
maybe can try using verbs for methods like all your other functions. 
(small things filling quota :P)

nictjh commented on src/main/java/nicholas/ui/Storage.java:
```diff
@@ -0,0 +1,79 @@
+     * @param filePath The path of the file to be emptied.
+     * @throws IOException If an error occurs while emptying the file.
+     */
+    public static void emptyFile(String filePath) throws IOException {
```
Similar comments, from above, can try using verbs, maybe "clearFileContents"?

nictjh commented on src/test/java/nicholas/ParserTest.java:
```diff
@@ -0,0 +1,35 @@
+public class ParserTest {
+
+    @Test
+    void testParseCommand() {
```
Quote the coding standards > 
Underscores may be used in test method names using the following three-part format featureUnderTest_testScenario_expectedBehavior()

nictjh reviewed (commented):
solid naming conventions bro
solid work

nicholasdaijh committed:
Add GUI

nicholasdaijh committed:
Fix long methods

nicholasdaijh committed:
Enable assertions

This changes the default setting of assertions being false to true

nicholasdaijh committed:
Add assertions

As a way to ensure that user inputs are valid, let's add some assertions to define assumptions about the program state so that the runtime can verify them.

nicholasdaijh committed:
Improve code quality

nicholasdaijh committed:
Change tasks.txt

nicholasdaijh committed:
Merge branch 'branch-A-Assertions'

* branch-A-Assertions:
  Add assertions
  Enable assertions

# Conflicts:
#	tasks.txt

nicholasdaijh committed:
Add task priority

nicholasdaijh committed:
Add ui.png and user guide

nicholasdaijh committed:
Update README.md

nicholasdaijh committed:
Remove background image for GUI

nicholasdaijh committed:
Merge commit 'f1bfd473b79c393c2d1411e8da0db4dac6b9be7b'

* commit 'f1bfd473b79c393c2d1411e8da0db4dac6b9be7b':
  Update README.md

nicholasdaijh committed:
Change Ui.png

nicholasdaijh committed:
Fix exception handling
